datashader/reductions.py

```diff
@@ -875,20 +875,22 @@ class first(Reduction):
     _dshape = dshape(Option(ct.float64))
 
     @staticmethod
-    def _append(x, y, agg):
-        raise NotImplementedError("first is currently implemented only for rasters")
+    @ngjit
+    def _append(x, y, agg,field):
+        if not isnull(field) and isnull(agg[y, x]):
+            agg[y, x] = field
 
     @staticmethod
     def _create(shape, array_module):
-        raise NotImplementedError("first is currently implemented only for rasters")
+        return array_module.full(shape, np.nan)
 
     @staticmethod
     def _combine(aggs):
         raise NotImplementedError("first is currently implemented only for rasters")
```

Comment:
This `NotImplementedError` needs a different explanation now, perhaps "first is not implemented for dask DataFrames"?